### PR TITLE
Issue #2169: Update HTML documentation to describe all options for all checks

### PR DESCRIPTION
**Pull request by MEZk**

### CODE REVIEW

**Comment by romani** [comment]
Please share new website on github.io , this will speadup merge process

File: src/xdocs/config_blocks.xml [hunk 1]
```diff
@@ -50,31 +50,36 @@
...
+              <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_CASE">LITERAL_CASE</a>,
+              <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_DEFAULT">LITERAL_DEFAULT</a>,
+              <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ARRAY_INIT">ARRAY_INIT</a>,
+              <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_CATCH">LITERAL_CATCH</a>.
+            </td>
+            <td>
+              <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_WHILE">LITERAL_WHILE</a>,
+              <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_TRY">LITERAL_TRY</a>,
+              <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_FINALLY">LITERAL_FINALLY</a>,
+              <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_DO">LITERAL_DO</a>,
```
Comment (mkordas) [inline src/xdocs/config_blocks.xml]: `LITERAL_DO` is here on 4th position, but in previous column it's on first. Shouldn't it be unified?

**Comment by mkordas** [comment]
@MEZk, changes are really impressive. Did you have any tool to generate them? Ideally we should have some tests to confirm that documentation is up-to-date.

**Comment by romani** [comment]
@MEZk , it is not clear why not all Checks segments are changed. I expect update to all Checks even that do not have any option right now at xdoc.

**Comment by MEZk** [comment]
@romani
For other checks acceptable token sets are equal to required token sets . User has no possibility to configure tokes for a check

**Comment by MEZk** [comment]
@mkordas 
No, it was done manually) But UT which checks that all Checks are present in xdoc will be implemented in accordance with https://github.com/checkstyle/checkstyle/issues/819

File: src/xdocs/config_blocks.xml (hunk 1 above)
Comment (MEZk) [inline src/xdocs/config_blocks.xml]: @mkordas 
Corrected.

**Comment by MEZk** [comment]
@romani 
http://mezk.github.io/site/index.html

**Comment by mkordas** [comment]
@MEZk 
Can you review carefully all the below checks? I noticed in all of them some kind of problems (eg.missing commas, not saying this is subset, not having two first columns etc.)
http://mezk.github.io/site/config_naming.html#LocalFinalVariableName
http://mezk.github.io/site/config_naming.html#LocalVariableName
http://mezk.github.io/site/config_coding.html#InnerAssignment
http://mezk.github.io/site/config_coding.html#MagicNumber
http://mezk.github.io/site/config_javadoc.html#AtclauseOrder
http://mezk.github.io/site/config_whitespace.html#NoWhitespaceAfter
http://mezk.github.io/site/config_whitespace.html#TypecastParenPad
http://mezk.github.io/site/config_coding.html#IllegalType

**Comment by MEZk** [comment]
@mkordas 
>http://mezk.github.io/site/config_naming.html#LocalFinalVariableName
http://mezk.github.io/site/config_naming.html#LocalVariableName
http://mezk.github.io/site/config_coding.html#InnerAssignment
http://mezk.github.io/site/config_coding.html#MagicNumber
http://mezk.github.io/site/config_javadoc.html#AtclauseOrder
http://mezk.github.io/site/config_whitespace.html#NoWhitespaceAfter
http://mezk.github.io/site/config_coding.html#IllegalType

Corrected.

http://mezk.github.io/site/config_whitespace.html#TypecastParenPad
Removed 'token' option because acceptable tokens set is equal to required tokens set. User can not configure tokens set for the check.

**Comment by romani** [comment]
merged as FF

**Pull request closed without merging**